pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -621,6 +621,11 @@ type CommonPrometheusFields struct {
 	// If not specified, the configuration is reloaded using the /-/reload HTTP endpoint.
 	// +optional
 	ReloadStrategy *ReloadStrategyType `json:"reloadStrategy,omitempty"`
+
+	// Defines the maximum time that the `prometheus` container's startup probe will wait before being considered failed.
+	// If unset, it will be equal to 600 seconds (= 15 minutes).
+	// +optional
+	MaximumStartupDurationSeconds *int32 `json:"maximumStartupDurationSeconds,omitempty"`
```

Comment:
A few more edge cases to consider:
* what if we have a zero value?
* what if we have a negative value?

For simplicity, we could consider that it's equivalent to the value being nil. Or consider that 0 disables the probe?

Reply:
I prefer keep the default behavior if use set to zero we can use the default probe we have.

Reply:
I'm wondering if we should have a min value of 60 for example, to ensure it will never try to divide by 0

Reply:
Hmm not sure to follow. If we define the increase "step" to be 10s then any value here between 1 and 10 will be translated to 10 (and [11,20) -> 20, ...).

Reply:
Sorry!
I don't get you.
Could you come again?